src/cita/zoteroOverlay.tsx

```diff
@@ -458,16 +462,48 @@ class ZoteroOverlay {
 		}
 	}
 
-	getFromCrossref(menuName: MenuSelectionType) {
+	async getFromCrossref(menuName: MenuSelectionType) {
 		// get items selected
 		// filter items with doi
 		// generate batch call to crossref
 		// only add items not available locally yet
-		Crossref.getCitations();
+		const items = await this.getSelectedItems(menuName, true);
+		if (items.length) {
+			new Crossref().addCitationsToItems(items);
```

Comment:
That was my first try but from my (very limited and recent) TypeScript understanding, you can't enforce static functions in abstract classes. So if we want static, we lose the abstraction. I might be wrong though!

Reply:
Yeah, looks like you're right: https://github.com/microsoft/TypeScript/issues/34516

Maybe this would work with an interface instead?

Reply:
Again, my understanding is very limited, but enforcing static methods with interfaces also doesn't seem trivial. It does rub me the wrong way to have to initialize these Indexers instead of relying on static methods, but it seems quite lightweight anyway (no stored variables) and allows us to have clearer code.

Reply:
Yeah, I'll see if I can find a proper / easy way to do this. Otherwise like you say it isn't a big deal.